hatchet_sdk/clients/workflow_listener.py

```diff
@@ -159,14 +163,18 @@ async def _request(self) -> AsyncIterator[SubscribeToWorkflowRunsRequest]:
         while True:
             request = await self.requests.get()
 
-            if request is False or self.stop_signal:
-                self.stop_signal = True
+            # if the request is an int which matches the current requester, then we should stop
+            if request == self.curr_requester:
```

Comment:
This removes the possibility of race conditions related to pushing `False` into the queue. I think we were previously leaking `_request` methods because each `_retry_subscribe` wasn't being cleaned up.